pkg/webhook/resources/upgrade/validator.go

```diff
@@ -65,5 +85,64 @@ func (v *upgradeValidator) Create(request *types.Request, newObj runtime.Object)
 		return werror.NewConflict(msg)
 	}
 
+	if newUpgrade.Annotations != nil {
+		if skipWebhook, ok := newUpgrade.Annotations[skipWebhookAnnotation]; ok && strings.ToLower(skipWebhook) == "true" {
+			return nil
+		}
+	}
+
+	hasDegradedVolume, err := v.hasDegradedVolume()
+	if err != nil {
+		return werror.NewInternalError(err.Error())
+	}
+
+	if hasDegradedVolume {
+		return werror.NewBadRequest("there are degraded volumes, please check all volumes are healthy")
+	}
+
+	return nil
+}
+
+func (v *upgradeValidator) hasDegradedVolume() (bool, error) {
+	nodes, err := v.nodes.List(labels.Everything())
+	if err != nil {
+		return false, err
+	}
+
+	if len(nodes) < 3 {
+		return false, nil
+	}
+
+	volumes, err := v.lhVolumes.List(util.LonghornSystemNamespaceName, labels.Everything())
+	if err != nil {
+		return false, err
+	}
+
+	for _, volume := range volumes {
+		if volume.Status.Robustness == longhornv1beta1.VolumeRobustnessDegraded {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func (v *upgradeValidator) Delete(_ *types.Request, oldObj runtime.Object) error {
+	oldUpgrade := oldObj.(*v1beta1.Upgrade)
+	if oldUpgrade.Annotations != nil {
+		if skipWebhook, ok := oldUpgrade.Annotations[skipWebhookAnnotation]; ok && strings.ToLower(skipWebhook) == "true" {
+			return nil
+		}
+	}
+
+	cluster, err := v.clusters.Get(util.FleetLocalNamespaceName, util.LocalClusterName)
+	if err != nil {
+		return werror.NewInternalError(fmt.Sprintf("can't find %s/%s cluster, err: %+v", util.FleetLocalNamespaceName, util.LocalClusterName, err))
+	}
+
+	if cluster.Status.Phase == string(clusterv1alpha4.ClusterPhaseProvisioning) {
```

Comment:
seems better to check `ClusterPhaseProvisioned` instead of `ClusterPhaseProvisioning`

Reply:
@FrankYang0529 It's a good idea, do you think we can change it?

Reply:
Yeah, I think we can change it to `if cluster.Status.Phase != string(clusterv1alpha4.ClusterPhaseProvisioned)`. I will update it in further PR.

Reply:
@FrankYang0529 @bk201 
Again, 
the Rancher cluster status check is added in `func (v *upgradeValidator) Delete`,  should we also add it to `(v *upgradeValidator) Create` ?   the wait cluster ready is done in shell script, and the validator should check it at first, right?

Reply:
It's added in https://github.com/harvester/harvester/pull/3093